testify/test_case.py

```diff
@@ -195,6 +195,12 @@ def runnable_test_methods(self):
     def run(self):
         """Delegator method encapsulating the flow for executing a TestCase instance.
         """
+        # The TestResult constructor wants an actual method, which it inspects
+        # to determine the method name (and class name, so it must be a method
+        # and not a function!). self.run is as good a method as any.
+        test_case_result = TestResult(self.run)
+        test_case_result.start()
+        self.fire_event(self.EVENT_ON_RUN_TEST_CASE, test_case_result)
```

Comment:
Fixed flake8 failures and also added new tests which will do basic check to avoid breaking it in future.

We can write an integration tests in another change because it will require time and this current fix needs to be pushed asap. Created the ticket for this https://github.com/Yelp/Testify/issues/323.

We need the fake .run for now because removing it is breaking TestResults. I will create a ticket for this once current changes are pushed.